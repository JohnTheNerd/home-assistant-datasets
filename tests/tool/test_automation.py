"""Test for running the automation eval end to end."""

import logging
import pathlib
import subprocess
import tempfile

from syrupy import SnapshotAssertion


_LOGGER = logging.getLogger(__name__)


def run_cmd(cmds: list[str]) -> None:
    """Run the specified commands."""
    _LOGGER.debug(cmds)
    p = subprocess.Popen(cmds, stdout=subprocess.PIPE)
    (cmd_output, _) = p.communicate()
    _LOGGER.debug("Output: %s", cmd_output)
    return p.returncode


def test_automation(snapshot: SnapshotAssertion) -> None:
    """Run the automation eval."""
    with tempfile.TemporaryDirectory() as tmpdir:
        run_cmd(
            [
                "home-assistant-datasets",
                "automation",
                "collect",
                "--models=assistant",
                "--dataset=datasets/automations",
                f"--model_output_dir={tmpdir}",
                "--categories=light",
                "--count=1",
            ]
        )
        # Ensure the scrape succeeded before proceeding
        assert (pathlib.Path(tmpdir) / "assistant" / "_scrape_context.yaml").exists()

        run_cmd(
            [
                "home-assistant-datasets",
                "automation",
                "eval",
                "--dataset=datasets/automations",
                f"--model_output_dir={tmpdir}",
            ]
        )

        output_files = sorted(list(pathlib.Path(tmpdir).glob("*.*")))
        assert [output.name for output in output_files] == snapshot
        for output in output_files:
            assert (output.name, output.read_text()) == snapshot
