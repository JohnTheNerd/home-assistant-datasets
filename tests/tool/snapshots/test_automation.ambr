# serializer version: 1
# name: test_automation
  list([
    'report.csv',
    'reports-by-task.yaml',
    'reports-token-stats.yaml',
    'reports.yaml',
  ])
# ---
# name: test_automation.1
  tuple(
    'report.csv',
    '''
      task_id,model_id,task_name,label,details
      "light_on_door_light_on_door-light_on_door-0","assistant","light_on_door-test_blueprint_inputs","Bad","AssertionError: Blueprint content is not available"
      "light_on_door_light_on_door-light_on_door-0","assistant","light_on_door-test_door_open","Bad","AssertionError: Could not extract YAML from model response: Sorry, I am not aware of any area called # Light on door ## Problem statement Create an blueprint au..."
      "light_on_door_light_on_door-light_on_door-0","assistant","light_on_door-test_door_open_close","Bad","AssertionError: Could not extract YAML from model response: Sorry, I am not aware of any area called # Light on door ## Problem statement Create an blueprint au..."
      "light_on_door_light_on_door-light_on_door-0","assistant","light_on_door-test_light_timeout","Bad","AssertionError: Could not extract YAML from model response: Sorry, I am not aware of any area called # Light on door ## Problem statement Create an blueprint au..."
  
    ''',
  )
# ---
# name: test_automation.2
  tuple(
    'reports-by-task.yaml',
    '''
      ---
      - task_id: light_on_door_light_on_door-light_on_door-0
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 4
      
  
    ''',
  )
# ---
# name: test_automation.3
  tuple(
    'reports-token-stats.yaml',
    '''
      --- []
      
  
    ''',
  )
# ---
# name: test_automation.4
  tuple(
    'reports.yaml',
    '''
      ---
      - model_id: assistant
        good_percent: 0.0%
        confidence_interval: 0.0%
        good: 0
        total: 4
      
  
    ''',
  )
# ---
